project/Build.scala

```diff
@@ -16,42 +17,52 @@ object Build {
-    "-Xlint:type-parameter-shadow",  // A local type parameter shadows a type already in scope.
-    "-Xlint:delayedinit-select",     // Selecting member of DelayedInit.
-    "-Xlint:stars-align",            // Pattern sequence wildcard must align with sequence component.
-    "-Xlint:option-implicit",        // Option.apply used implicit view.
-    "-Xlint:poly-implicit-overload", // Parameterized overloaded implicit methods are not visible as view bounds.
-    "-Ywarn-extra-implicit"          // Warn when more than one implicit parameter section is defined.
-  ) ++
+  lazy val ScalacOptions = Def.setting(
     Seq(
-      "-Ywarn-unused:imports",       // Warn if an import selector is not referenced.
-      "-Ywarn-unused:locals",        // Warn if a local definition is unused.
-      "-Ywarn-unused:privates",      // Warn if a private member is unused.
-      "-Ywarn-unused:implicits"      // Warn if an implicit parameter is unused.
-    ).filter(_ => shouldWarnForUnusedCode)
+      "-encoding",
+      "UTF-8",
+      "-unchecked",
+      "-deprecation",
+      "-feature",
+      "-language:postfixOps",
+      "-language:implicitConversions",
+      "-language:higherKinds",
+      "-Xfatal-warnings",
+      "-Xlint:nullary-unit",           // Warn when nullary methods return Unit.
+      "-Xlint:inaccessible",           // Warn about inaccessible types in method signatures.
+      "-Xlint:missing-interpolator",   // A string literal appears to be missing an interpolator id.
+      "-Xlint:doc-detached",           // A Scaladoc comment appears to be detached from its element.
+      "-Xlint:private-shadow",         // A private field (or class parameter) shadows a superclass field.
+      "-Xlint:type-parameter-shadow",  // A local type parameter shadows a type already in scope.
+      "-Xlint:delayedinit-select",     // Selecting member of DelayedInit.
+      "-Xlint:stars-align",            // Pattern sequence wildcard must align with sequence component.
+      "-Xlint:option-implicit",        // Option.apply used implicit view.
+      "-Xlint:poly-implicit-overload", // Parameterized overloaded implicit methods are not visible as view bounds.
+      "-Ywarn-extra-implicit"          // Warn when more than one implicit parameter section is defined.
+    ) ++
+      Seq(
+        "-Ywarn-unused:imports",       // Warn if an import selector is not referenced.
+        "-Ywarn-unused:locals",        // Warn if a local definition is unused.
+        "-Ywarn-unused:privates",      // Warn if a private member is unused.
+        "-Ywarn-unused:implicits"      // Warn if an implicit parameter is unused.
+      ).filter(_ => shouldWarnForUnusedCode)
+      ++ (CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, n)) if n <= 12 =>
+          Seq()
+        case _                       =>
+          Seq(
+            "-Ymacro-annotations"

```

Comment:
It seems that macro is not used in this project.
If so, this pattern matching for `"-Ymacro-annotations"` can be removed.

Reply:
You're right, it's not needed. Looks like it was another thing that got included since I copy/pasted this build.sbt from another project. I can remove it.